demos/2019_RHTE/labs/7.md

```diff
@@ -0,0 +1,268 @@
+# 7.0 Debugging Failed Migrations
+
+Most of the time migrations go as planned, but knowing what to do when they don't is critical.  This lab attempts to provide you with a process/guide to follow when investigating a failed migration.
+
+![Failed Migration](./screenshots/lab7/mig-plan-failed.png)
+
+## 7.1 CRD Architecture
+
+![Failed Migration](./screenshots/lab7/mig-custom-resources.png)
+
+It's important to first understand the custom resources in play.  The above architecture diagram illustrates the migration resources and their relationships.  Most of our debugging failed migration executions will focus on the right-side of the diagram:
+ * MigMigration
+ * Velero - Backup and Restore
+
+## 7.1 Workflow
+
+Upon execution of a Migration Plan (stage or migrate), a `MigMigration` is created.  This custom resource is created for each distinct run; and is created on the same cluster where the migration-controller that is orchestrating the migration is running.
+
+Let's take a look at the existing MigMigrations.  On our 4.1 cluster, perform the following:
+
+```bash
+oc get migmigration -n mig
+NAME                                   AGE
+88435fe0-c9f8-11e9-85e6-5d593ce65e10   6m42s
+```
+Once you'd found the appropriate migration execution that you want to investigate, you can request more detail:
```

Comment:
Might want to reword to something like below

"Once you find the appropriate migration execution that you want to investigate, you can request more detail"